bundles/core/src/test/java/com/adobe/cq/commerce/core/components/internal/models/v1/productteaser/ProductteaserImplTest.java

```diff
@@ -0,0 +1,103 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+package com.adobe.cq.commerce.core.components.internal.models.v1.productteaser;
+
+import com.adobe.cq.commerce.core.components.models.product.Asset;
+import com.adobe.cq.commerce.core.components.service.ProductGraphqlService;
+import com.adobe.cq.commerce.magento.graphql.*;
+import com.day.cq.wcm.api.Page;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.internal.util.reflection.Whitebox;
+
+import java.text.NumberFormat;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Locale;
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class ProductteaserImplTest {
+
+    ProductteaserImpl slingModel;
+
+    private static final String TEST_URL="/testUrl/a/b";
+    private static final String TEST_SLUG="dummyslug";
+    private static final String TEST_PRODUCTNAME="TESTPRODUCT";
+    private static final String TEST_PRODUCTPRICE="$400.00";
+
+
+    @Before
+    public void setup() {
+        this.slingModel = new ProductteaserImpl();
+        ProductInterface gqlProduct = mock(ProductInterface.class);
```

Comment:
I think the unit tests of `ProductImplTest` are not very good examples of how we should do these unit tests, we should change that. Look at `ProductListImplTest`, I think it's way better and uses some JSON response instead of manually mocking all java objects.